Userland/Libraries/LibJS/Runtime/GlobalObject.cpp

```diff
@@ -226,10 +227,20 @@ JS_DEFINE_NATIVE_FUNCTION(GlobalObject::parse_float)
         return vm.argument(0);
     auto input_string = TRY(vm.argument(0).to_string(vm));
     auto trimmed_string = MUST(trim_string(vm, js_string(vm, input_string), TrimMode::Left));
-    for (size_t length = trimmed_string.length(); length > 0; --length) {
-        auto number = MUST(Value(js_string(vm, trimmed_string.substring(0, length))).to_number(vm));
-        if (!number.is_nan())
-            return number;
+    auto result = parse_first_floating_point<double>(trimmed_string.characters(), trimmed_string.characters() + trimmed_string.length());
+    if (result.parsed_value())
+        return result.value;
+    if (trimmed_string.is_empty())
```

Comment:
if the trimmed string is empty, wouldn't we want to check that before trying to parse it as a float? :thinking:

Reply:
Yeah I guess that's better, I think I put the empty check here to be able to safely use `trimmed_string[0]` below